### DISCUSSION TITLE
Alpine v3 doesn't execute directives inside nested templates with x-if

### DISCUSSION THREAD

**Discussion by Niklan** [alpinejs/alpine#1535] [7. Bugs (archived)] [answered]
It's simple as that:

```html
<div x-data="{items: [{id: 1, label: '1'}, {id: 2, 'label': '2'}]}">
  
  <template x-for="item in items" :key="item.id">
    <div>
      Item label: 
      <template x-if="item.label">
        <span x-text="item.label"></span>
      </template>
    </div>
  </template>
</div>
```

And it doesn't work. 🤷‍♂️

The `<span x-text="item.label"></span>` is added to the page, means that template is working, but not execute directives on new DOM elements. If we move  `<span x-text="item.label"></span>` out of `x-if` it does works.

* **Alpine v3** https://codepen.io/Niklan/pen/XWMOxKa - example from above
* **Alpine v2**: https://codepen.io/Niklan/pen/yLMZRoM - working as expected

**Comment by Niklan** [comment, marked as answer]
There is Pull Request that possible solves the problem https://github.com/alpinejs/alpine/pull/1536

**Reply by calebporzio** [reply]
This has been merged in, refactored and tested - next release will have this fixed. Thanks!
